test/jit/test_graph_rewrite_passes.py

```diff
@@ -4,58 +4,149 @@
 
         # check matmuls are not fused
-        class Matmul(torch.nn.Module):
-            def __init__(self, weight):
-                super(Matmul, self).__init__()
-                self.weight = weight
-
-            def forward(self, x):
-                return torch.matmul(x, self.weight)
-
-        x = torch.rand(5, 6, 5)
-        w = torch.rand(5, 5, 100)
-        model = torch.jit.trace(Matmul(w), [x])
-        torch._C._jit_pass_fuse_linear(model.graph)
-        # check 3d matmul is not fused
-        FileCheck().check("aten::matmul").run(model.graph)
-        FileCheck().check_not("aten::linear").run(model.graph)
-        # make sure it runs
-        model(x)
+        x_3 = torch.rand(5, 6, 5)
+        w_3 = torch.rand(5, 5, 100)
+        model_3 = torch.jit.trace(FunctionalMatmul(w_3), [x_3])
+        check_pattern_count_map_3 = {"aten::linear": 0}
+        self.check_single_replacement("aten::matmul", "aten::matmul", torch._C._jit_pass_fuse_linear, model_3)
+        self.check_op_presence(check_pattern_count_map_3, torch._C._jit_pass_fuse_linear, model_3)
+        model_3(x_3)  # make sure it runs
+
+    @unittest.skipUnless(environ.get("USE_VULKAN") == "1", "Test depends on Vulkan installation.")
+    def test_vulkan_insert_pre_packed_ops(self) -> None:
+        x_1 = torch.rand(3)
+        w_1 = torch.rand(5, 3)
+        b_1 = torch.rand(5)
+        model_1 = torch.jit.trace(FunctionalLinear(w_1, b_1), [x_1])
+        check_pattern_count_map_1 = {"aten::matmul": 0, "aten::add_": 0, "aten::t": 0}
+        self.check_single_replacement(
+            "aten::matmul",
+            "vulkan_prepack::linear_run",
+            torch._C._jit_pass_vulkan_insert_prepacked_ops,
+            model_1
+        )
+        self.check_op_presence(check_pattern_count_map_1, torch._C._jit_pass_vulkan_insert_prepacked_ops, model_1)
+        model_1(x_1)  # make sure it runs
+
+        conv2d_in_channels = 3
+        conv2d_out_channels = 4
+        conv2d_kernel = 2
+        conv2d_weight = torch.rand(
+            conv2d_out_channels,
+            conv2d_in_channels,
+            conv2d_kernel,
+            conv2d_kernel,
+        )
+        conv2d_bias = torch.rand(conv2d_out_channels)
+        x_2_shape = (3, 2, 5)
+        x_2 = torch.rand(x_2_shape)
+
+        model_2 = torch.jit.trace(FunctionalConv2d(conv2d_weight, conv2d_bias), [x_2])
+        check_pattern_count_map_2 = {
+            "aten::_convolution": 0,
+            "vulkan_prepack::conv2d_clamp_run": 1,
+            "vulkan_prepack::conv2d_clamp_prepack": 1
+        }
+        self.check_single_replacement(
+            "aten::_convolution",
+            "prim::Constant",
+            torch._C._jit_pass_vulkan_insert_prepacked_ops,
+            model_2
+        )
+        self.check_op_presence(
+            check_pattern_count_map_2,
+            torch._C._jit_pass_vulkan_insert_prepacked_ops,
+            model_2
+        )
+        model_2(x_2)  # make sure it runs
```

Comment:
(to execute the vulkan model) convert the input tensor to vulkan before executing, and convert the output tensor back to CPU.

```
out = model_2(x_2.to(device="vulkan").to(device="cpu")
```